scipy/linalg/decomp_cholesky.py

```diff
@@ -41,10 +41,23 @@ def _cholesky(a, lower=False, overwrite_a=False, clean=True,
     if info < 0:
         raise ValueError('LAPACK reported an illegal value in {}-th argument'
                          'on entry to "POTRF".'.format(-info))
-    return c, lower
 
+    # if the pivot flag is false, return the result
+    if not chol_pivot:
```

Comment:
This switch must happen earlier so you don't call both potrf and pstrf depending on the flag. You need to solve only once. Either one or the other.

Reply:
OK, fixed.